### PR TITLE
Remove: __experimentalHasMultipleOrigins prop from colors and gradients

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Part of https://github.com/WordPress/gutenberg/issues/42994.
This PR removes the __experimentalHasMultipleOrigins from all the color and gradient-related components.
Instead of having this property, we now inspect the contents of colors and gradients to understand its shape.

cc: @ciampo 

## ✍️ Dev Note

The `__experimentalHasMultipleOrigins` prop has been removed from the `ColorPalette`, `GradientPicker`, `BorderControl` and `BorderBoxControl` components in the `@wordpress/components` package.

The prop has been replaced by a check in the `ColorPalette` component which automatically detects whether the colors passed through the `colors` prop has a single or multiple color origins.

### CODE REVIEW

**Comment by t-hamano** [comment]
@jorgefilipecosta 

I came across this PR while researching #46148 regarding the `ColorGradientSettingsDropdown ` component.

The problem is that when `ColorGradientSettingsDropdown` is used directly, the UI is rendered even if the color settings are empty. An empty check may also be necessary in the `ColorGradientSettingsDropdown` component.

Should this be addressed in the follow-up PR?

File: packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js [hunk 1]
```diff
@@ -140,10 +164,7 @@ const PanelColorGradientSettings = ( props ) => {
 	) {
 		return <PanelColorGradientSettingsInner { ...props } />;
 	}
-	if ( props.__experimentalHasMultipleOrigins ) {
```
Comment (youknowriad) [inline packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js:143]: I don't understand the changes on this component, can you detail a bit more?

For me you had two options here:

 - Either you could have forced all the controls to be "multiple" or "single" (you choose here the shape of the colors/gradients prop to provide to the lower level components from the components package)
 - Or keep the `__experimentalHasMultipleOrigins` prop at this level (block-editor package components) and construct the shape of the palettes depending on this prop, leaving the choice to the consumer of these components.

**Review by youknowriad — COMMENTED** [review]
Looks like there's some related e2e test failures.

**Comment by bph** [comment]
Added the 'needs dev note' label, so it's flagged to get a mention it in the 6.2 Fieldguide. 
[Seems 183 plugins](https://wpdirectory.net/search/01GM8GQ4Z0AG9Q56H1Q3773HZ8) are using the string

**Review by youknowriad — APPROVED** [review]
This LGTM, you might want to rebase the PR per the recent jobs changes.

**Comment by jorgefilipecosta** [comment]
> Added the 'needs dev note' label, so it's flagged to get a mention it in the 6.2 Fieldguide. [Seems 183 plugins](https://wpdirectory.net/search/01GM8GQ4Z0AG9Q56H1Q3773HZ8) are using the string

Hi @bph, we can write a dev note to get more awareness and testing. But with this PR, users of the string __experimentalHasMultipleOrigins will not notice any change. Our code just got smarter and detects when __experimentalHasMultipleOrigins behavior should be used or not without developers needing to pass it.

**Comment by ciampo** [comment]
Sorry, catching up through a long queue 😅 

Great cleanup work @jorgefilipecosta ! I opened a follow-up PR (https://github.com/WordPress/gutenberg/pull/47384) with a few more cleanup tweaks, but nicely done here 🙌 

It would be great to see the same treatment applied to the `__experimentalIsRenderedInSidebar` prop

**Pull request merged**